src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -0,0 +1,575 @@
+	assert(data.userId, "AnalyticsService.LogMarketplacePurchase - userId is required")
+	assert(data.itemType, "AnalyticsService.LogMarketplacePurchase - itemType is required")
+	assert(data.id, "AnalyticsService.LogMarketplacePurchase - id is required")
+	assert(data.cartType, "AnalyticsService.LogMarketplacePurchase - cartType is required")
+	assert(data.userId, "AnalyticsService.LogMarketplacePurchase - userId is required")
+	assert(
+		data.amount == nil or typeof(data.amount) == "number",
+		"AnalyticsService.LogMarketplacePurchase - amount must be a number"
+	)
+	
+	GameAnalytics:addBusinessEvent(data.userId, {
+		itemType = data.itemType,
+		itemId = data.id,
+		amount = data.amount or 1,
+		cartType = data.cartType
+	})
+end
+
+--[=[
+	Shortcut function for LogResourceEvent
+	Used to log in-game currency purchases
+	
+	Example Use:
+	```lua
+	AnalyticsService:LogPurchase({
+		userId = player.UserId,
+		eventType = "Shop",
+		currency = "Coins",
+		itemId = "Red Paintball Gun"
+	})
+	```
+]=]
+function AnalyticsService:LogPurchase(data: PurchaseEvent)
+	assert(typeof(data) == "table", "AnalyticsService.LogPurchase - data is required")
+	assert(typeof(data.userId) == "number", "AnalyticsService.LogPurchase - userId is required")
+	assert(typeof(data.eventType) == "string", "AnalyticsService.LogPurchase - eventType is required")
+	assert(
+		table.find(RESOURCE_EVENT_TYPES, data.eventType),
+		"AnalyticsService.LogPurchase - eventType " .. data.eventType .. " is invalid. Please define it in RESOURCE_EVENT_TYPES"
+	)
+	assert(data.itemId, "AnalyticsService.LogPurchase - itemId is required")
+	assert(data.currency, "AnalyticsService.LogPurchase - currency is required")
+	assert(
+		table.find(CURRENCIES, data.currency),
+		"AnalyticsService.LogPurchase - currency type is invalid"
+	)
+	assert(
+		data.amount == nil or typeof(data.amount) == "number",
+		"AnalyticsService.LogPurchase - amount is required"
+	)
+	assert(
+		data.flowType == nil or GameAnalytics.EGAResourceFlowType[data.flowType],
+		"AnalyticsService.LogPurchase - flow type is invalid"
+	)
+	
+	self:LogResourceEvent({
+		userId = data.userId,
+		amount = data.amount or 1,
+		currency = data.currency,
+		flowType = (
+			data.flowType == GameAnalytics.EGAResourceFlowType.Source and
+			GameAnalytics.EGAResourceFlowType.Source
+		) or (
+			data.flowType == GameAnalytics.EGAResourceFlowType.Sink and
+			GameAnalytics.EGAResourceFlowType.Sink
+		) or
+			GameAnalytics.EGAResourceFlowType.Sink,
+		eventType = data.eventType,
+		itemId = data.itemId
+	})
+end
+
+--[=[
+	Used to log in-game currency changes (example: player spent coins in a shop,
+	player purchased coins, player won coins in a mission)
+	
+	Example Use:
+	```lua
+	-- Player purchased 100 coins with Robux
+	AnalyticsService:LogPurchase({
+		userId = player.UserId,
+		eventType = "Purchase",
+		currency = "Coins",
+		itemId = "100 Coins"
+	})
+	```
+]=]
+function AnalyticsService:LogResourceEvent(data: ResourceEvent)
```

Comment:
Add param tag to the moonwave block.
@param data ResourceEvent -- ResourceEvent type

Reply:
See earlier note about using promise instead of assert